src/cluster_legacy.c

```diff
@@ -4960,10 +4966,22 @@ void clusterHandleManualFailover(void) {
  * CLUSTER cron job
  * -------------------------------------------------------------------------- */
 
+static mstime_t getHandshakeTimeout(void) {
+    /* The handshake timeout is the time after which a handshake node that was
+     * not turned into a normal node is removed from the nodes. Usually it is
+     * just the cluster_node_timeout value, but when cluster_node_timeout is
+     * too small we use the value of 1 second. */
+    return max(server.cluster_node_timeout, 1000);
+}
+
+static int nodeExceedsHandshakeTimeout(clusterNode *node, mstime_t now) {
+    return now - node->ctime > getHandshakeTimeout() ? 1 : 0;
```

Comment:
Should we use `inbound_link_freed_time` instead? also the field name isn't super clear IMO. Can we consider reusing`meet_sent`? We should also move the reset of this time out of `freeClusterLink`.

```suggestion
    return now - node-> meet_sent > getHandshakeTimeout() ? 1 : 0;
```

Reply:
We can't use `meet_sent` here as this function is also used by the receiver of the MEET packet to check if the handshake timed out.

Reply:
> also the field name isn't super clear IMO.

Would you prefer something like `inbound_link_last_freed_time`?